pkg/operator/ceph/spec/spec.go

```diff
@@ -90,20 +90,17 @@ func confGeneratedInPodVolumeAndMount() (v1.Volume, v1.VolumeMount) {
 }
 
 // PodVolumes fills in the volumes parameter with the common list of Kubernetes volumes for use in Ceph pods.
-// This function is only used for OSDs.
+// This function is only used for OSDs. dataDirHostPath MUST be specified.
 func PodVolumes(dataPaths *config.DataPathMap, dataDirHostPath string, confGeneratedInPod bool) []v1.Volume {
 
-	dataDirSource := v1.VolumeSource{EmptyDir: &v1.EmptyDirVolumeSource{}}
-	if dataDirHostPath != "" {
```

Comment:
Remind me why we need to require the dataDirHostPath? Seems like a good thing to remove being tied to the hosts. If someone configured cluster-on-pvc.yaml and didn't set the dataDirHostPath, their cluster wouldn't need the host path anymore. The main downside is that you couldn't collect the crash logs.

Reply:
Great question . . . maybe this is why I dropped this PR a couple months ago.

This has to be set unless we have both mons and osds on PVCs I would think. Does that logic make sense? We could make that be the check here, and we could still allow an emptydir to be used for whole-cluster-on-PVC; at the same time, I think we would have to disable the crashcontroller if Rook can't send logs to hosts. 

This sounds more tricky than I'd like. I wonder if, short term, we should disallow the dataDirHostPath from being empty (this PR), and long term we change the crash controller to support emptyDirs for the whole-cluster-on-PVC case.

As for external cluster concerns, @leseb can answer better I think.

Reply:
I'm inclined to vote that we don't need this PR. Seems like it's already supporting the cluster-on-pvc scenario where the dataDirHostPath is not set. If that's already working, why would we prevent it?